### PR TITLE
a few improvement for addr_adv path manager

### PR DESCRIPTION
**Pull request by pabeni**
this pull requests implements a few improvement for the addr_adv path manager:

- mptcpd can specify non 0 flags for the created endpoint, allowing egress subflow creation and/or ADD_ADDR signaling
- make address notification configurable, allow filtering link locak and loopback addresses and allowing processing addresses available at daemon startup
- set suitable limits for the configured endpoints

### CODE REVIEW

**Comment by ossama-othman** [comment]
I committed changes with the following updates:

* documentation and coding style updates, including `const` correctness and placement
* type correctness, i.e. `strlen()` returns `size_t`, not `int`
* unit test build fixes related to the move of `src/configuration.h` to `include/mptcpd/private/configuration.h`

With the unit test build fixes in place, the test suite now shows that the network monitor test crashes with a seg fault:

```c
#0  0x00007ffff7fb9291 in handle_rtm_getaddr () from .../addr_add/lib/.libs/libmptcpd.so.1
#1  0x00007ffff7f5b6e3 in process_multi (nlmsg=0x7fffffffcbcc, netlink=0x5555555597a0) at ell/netlink.c:221
#2  can_read_data (io=<optimized out>, user_data=0x5555555597a0) at ell/netlink.c:287
#3  0x00007ffff7f59610 in io_callback (fd=<optimized out>, events=1, user_data=0x555555559810) at ell/io.c:120
#4  0x00007ffff7f587ad in l_main_iterate (timeout=<optimized out>) at ell/main.c:478
#5  0x00007ffff7f5887c in l_main_run () at ell/main.c:525
#6  l_main_run () at ell/main.c:507
#7  0x00005555555553af in main ()
```
Valgrind shows an `Invalid read of size 1`.  I'll take a closer look in the morning.

The crash doesn't happen on the `master` branch.

File: src/configuration.c [hunk 1]
```diff
@@ -169,6 +322,12 @@ static struct argp_option const options[] = {
           0,
           "Log to DEST (stderr, syslog or journal), e.g. --log=journal",
           0 },
+        { "notify-flags",
+          MPTCPD_NOTIFY_FLAGS_KEY,
+          "BOOL",
```
Comment (ossama-othman) [inline src/configuration.c:327]: Shouldn't this be `FLAGS`?

**Review by ossama-othman — COMMENTED** [review]
Overall, the changes look good.  We should be good to go once we take care of the `test-network-monitor` unit test crash.

File: src/configuration.c [hunk 2]
```diff
@@ -169,6 +322,12 @@ static struct argp_option const options[] = {
...
+          0,
+          "FLags for address notify notification, e.g. --notify-flags=existing,skip_link_local,skip_loopback",
```
Comment (ossama-othman) [inline src/configuration.c:329]: Shorten to `"Address notification flags, e.g. --notify-flags=existing,skip_link_local,skip_loopback"`?

Comment (pabeni) [reply src/configuration.c:329]: Agreed on the shorten form.

'BOOL' is left-over from a previous internal version 'FLAGS' should be more accurate

**Review by ossama-othman — APPROVED** [review]
@pabeni Thanks for the improvements and bug fixes!  Much appreciated.  Everything looks good!

**Pull request merged**